pkg/apis/pipeline/v1alpha1/pipeline_validation.go

```diff
@@ -125,5 +124,33 @@ func (ps *PipelineSpec) Validate() *apis.FieldError {
 		return apis.ErrInvalidValue(err.Error(), "spec.tasks.resources.inputs.from")
 	}
 
+	// The parameter variables should be valid
+	if err := validatePipelineParameterVariables(ps.Tasks, ps.Params); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validatePipelineParameterVariables(tasks []PipelineTask, params []PipelineParam) *apis.FieldError {
+	parameterNames := map[string]struct{}{}
+	for _, p := range params {
+		parameterNames[p.Name] = struct{}{}
+	}
+	return validatePipelineVariables(tasks, "params", parameterNames)
+}
+
+func validatePipelineVariables(tasks []PipelineTask, prefix string, vars map[string]struct{}) *apis.FieldError {
+	for _, task := range tasks {
+		for _, param := range task.Params {
+			if err := validatePipelineVariable(fmt.Sprintf("param[%s]", param.Name), param.Value, prefix, vars); err != nil {
+				return err
+			}
+		}
+	}
 	return nil
 }
+
+func validatePipelineVariable(name, value, prefix string, vars map[string]struct{}) *apis.FieldError {
+	return ValidateVariable(name, value, prefix, "", "task parameter", "pipelinespec.params", vars)
```

Comment:
`s/task parameter/pipeline parameter/`

Reply:
Nope, `task parameter` - this is the name of the field we’re looking in, not where the parameter came from originally.

Reply:
what is the value of this `validatePipelineVariable `? Looks like there is no value add so I would consider calling `ValidateVariable ` function directly from `validatePipelineVariables `. WDYT @abayer ?

Reply:
Preventing excessive verbosity, basically - `ValidateVariable` is generic enough to be used either here or in `task_validation`, but that means it's got to take a bunch of string as parameters that'll always be the same in every call in `pipeline_validation`, and will always be the same in every call in `task_validation`. So hey, helper function to avoid excess copy paste noise. =) I'm not going to insist on keeping this, it just seemed cleaner to me.